src/test_disk.rs

```diff
@@ -1179,9 +1181,57 @@ fn filter_disks(devices: &[PathBuf], storage_detail_id: u32) -> BynarResult<Vec<
                 }
             }
             true
-        }).collect();
+        })
+        .collect();
     Ok(filtered_devices)
 }
+
+// Add in any disks that the database knew about that linux can no longer find
+fn add_previous_devices(
+    devices: &mut Vec<BlockDevice>,
+    pool: &Pool<ConnectionManager>,
+    host_mapping: &HostDetailsMapping,
+) -> BynarResult<()> {
+    // Sometimes failed devices are removed from sys/udev and we can no
+    // longer find them. This will dig up previously known devices
+    let previously_known_devices = get_devices(&pool)?;
```

Comment:
needs to be called for a storage_detail_id
`get_devices(&pool, host_mapping.storage_detail_id)`